src/Provider/GenericProvider.php

```diff
@@ -141,7 +141,8 @@ protected function getRequiredOptions()
     private function assertRequiredOptions(array $options)
     {
         $missing = array_diff_key(array_flip($this->getRequiredOptions()), $options);
-        if ($missing) {
+
+        if (!empty($missing)) {
```

Comment:
Again, Scrutinizer is being over-zealous here:

> The expression `$missing` of type `array` is implicitly converted to a boolean; are you sure this is intended? If so, consider using `! empty($expr)` instead to make it clear that you intend to check for an array without elements.

It should be fine to leave as is, and we avoid a function call, if you want an optimization argument. ;-)